seekinternational/notification-library
# El 2791 notification library vulnerability node growl

What does this PR do?
---------------------
Removes "gulp-mocha" and "mocha-teamcity-reporter" libraries, since both of them depend on a vulnerable version of the "growl" library and updating "gulp-mocha" to its newest version breaks compatibility with our code coverage library ("gulp-istanbul").

As such, the "test.js" and "smoke.js" Gulp scripts have had their implementations changed. They now call updated versions of "Mocha" and "Nyc" CLI tools (replacing "gulp-mocha" and "gulp-istanbul" respectively) directly.

How can I test this?
--------------------
1. Checkout this repository
2. Execute ```npm install```
3. Execute ```npm run test```
4. The test should run normally, and you should see both tests results and code coverage results. 

Notes
-----
- Nyc is the official CLI tool for Istanbul
- a plugin setting has been added to the .babelrc for Nyc/Istanbul
